pandas/core/arrays/datetimelike.py

```diff
@@ -1674,7 +1674,7 @@ def validate_endpoints(closed):
     return left_closed, right_closed
 
 
-def validate_inferred_freq(freq, inferred_freq, freq_infer):
+def validate_inferred_freq(freq, inferred_freq, freq_infer, freq_inherit):
```

Comment:
this is pretty confusing, how about just allowing freq_infer to be True/False/None.

Reply:
so freq_infer=None would mean inherit but dont do active inference?  im not sure thats clearer

Reply:
updated with a simpler implementation